src/collections/Message/Message.js

```diff
@@ -1,66 +1,198 @@
+    useKeyOnly(icon, 'icon'),
+    useKeyOnly(hidden, 'hidden'),
+    useKeyOnly(visible, 'visible'),
+    useKeyOnly(floating, 'floating'),
+    useKeyOnly(compact, 'compact'),
+    useKeyOrValueAndKey(attached, 'attached'),
+    useKeyOnly(warning, 'warning'),
+    useKeyOnly(info, 'info'),
+    useKeyOnly(positive, 'positive'),
+    useKeyOnly(success, 'success'),
+    useKeyOnly(negative, 'negative'),
+    useKeyOnly(error, 'error'),
+    'message',
+    className,
+  )
 
-  static _meta = {
-    name: 'Message',
-    type: META.type.collection,
-  }
+  const dismissIcon = (dismissable || onDismiss) && <Icon name='close' onClick={onDismiss || _.noop} />
+  const rest = getUnhandledProps(Message, props)
 
-  render() {
-    const classes = classNames(
-      'ui',
-      this.props.className,
-      { icon: this.props.icon },
-      'message'
+  if (content || header || icon || list) {
+    return (
+      <div {...rest} className={classes}>
+        {dismissIcon}
+        {icon && iconPropRenderer(icon)}
+        {(header || content || list) && <MessageContent>
+          {header && <MessageHeader>{header}</MessageHeader>}
+          {list && <MessageList items={list} />}
+          {content && <p>{content}</p>}
+        </MessageContent>}
+      </div>
     )
+  }
 
-    const closeIcon = <Icon name='close' onClick={this.handleDismiss} />
-    const header = <Header>{this.props.header}</Header>
-    const icon = <Icon name={this.props.icon} />
+  return (
+    <div {...rest} className={classes}>
+      {dismissIcon}
+      {children}
+    </div>
+  )
+}
 
-    // wrap children in <p> if there is a header
-    const children = this.props.header ? <p>{this.props.children}</p> : this.props.children
+Message._meta = {
+  name: 'Message',
+  type: META.type.collection,
+  props: {
+    attached: ['bottom'],
+    color: sui.colors,
+    size: _.without(sui.sizes, 'medium'),
+  },
+}
 
-    // wrap header and children in content if there is an icon
-    const content = (
-      <div className='content'>
-        {this.props.header && header}
-        {children}
-      </div>
-    )
+Message.propTypes = {
+  /** Primary content of the message. */
+  children: customPropTypes.all([
+    PropTypes.node,
+    customPropTypes.mutuallyExclusive(['header', 'content']),
+    customPropTypes.givenProps(
+      { icon: PropTypes.node.isRequired },
+      customPropTypes.mutuallyExclusive(['icon'])
+    ),
+  ]),
 
-    const rest = getUnhandledProps(Message, this.props)
+  /** Additional classes. */
+  className: PropTypes.string,
 
-    return (
-      <div {...rest} className={classes} ref='message'>
-        {this.props.dismissable && closeIcon}
-        {this.props.icon && icon}
-        {this.props.icon && content}
-        {!this.props.icon && this.props.header && header}
-        {!this.props.icon && this.props.children && children}
-      </div>
-    )
-  }
+  /** Primary content.  Mutually exclusive with children. */
+  content: PropTypes.node,
```

Comment:
Since this content is wrapped in a `p` tag, should it be restricted to `string`, and if they need to control the markup children should be used instead?